### ISSUE TITLE
Can' connect to pubsub

### ISSUE THREAD

**Issue by mimi89999** [ChatSecure/ChatSecure-iOS#822]
```
Jul 28 21:46:42 lebihan.pl:tls	debug	pubsub.chatsecure.org is offering TLS, taking up the offer...
Jul 28 21:46:42 s2sout1cbc560	debug	sending: <starttls xmlns='urn:ietf:params:xml:ns:xmpp-tls'>
Jul 28 21:46:42 s2sout1cbc560	debug	Received[s2sout_unauthed]: <proceed xmlns='urn:ietf:params:xml:ns:xmpp-tls'>
Jul 28 21:46:42 lebihan.pl:tls	debug	Proceeding with TLS on s2sout...
Jul 28 21:46:42 socket	debug	server.lua: attempting to start tls on tcp{client}: 0xe35a38
Jul 28 21:46:42 c2sc7d8e8	debug	Received[c2s]: <a xmlns='urn:xmpp:sm:3' h='668'>
Jul 28 21:46:42 c2sc7d8e8	debug	#queue = 0
Jul 28 21:46:42 socket	debug	server.lua: ssl handshake error: sslv3 alert handshake failure
Jul 28 21:46:42 s2sout1cbc560	debug	s2s connection attempt failed: ssl handshake error: sslv3 alert handshake failure
Jul 28 21:46:42 s2sout1cbc560	info	Failed in all attempts to connect to pubsub.chatsecure.org
Jul 28 21:46:42 mod_s2s	debug	No other records to try for pubsub.chatsecure.org - destroying
Jul 28 21:46:42 s2sout1cbc560	debug	Destroying outgoing session lebihan.pl->pubsub.chatsecure.org: Connecting failed: ssl handshake error: sslv3 alert handshake failure
```

@chrisballinger Could you please investigate what's going on with the server?

**Comment by chrisballinger** [comment]
Possibly because your server requires a peer cert?

<img width="729" alt="screen shot 2017-07-28 at 1 15 08 pm" src="https://user-images.githubusercontent.com/518687/28734640-d17e13e0-7396-11e7-94de-4619298b609a.png">

**Comment by chrisballinger** [comment]
Here's our server reports:

* https://check.messaging.one/result.php?id=19246
* https://check.messaging.one/result.php?id=19248

**Comment by Zash** [comment]
If @mimi89999 s server takes issue with something, it would probably be the one aborting the connection. Here it's the chatsecure one aborting it. If you have logs that capture anything from the TLS negotiation, that's where I would look if I were you.

**Comment by Zash** [comment]
Attempting to connect with no non-forward-secret ciphers enabled, it fails. Using Prosody default ciphers, AES256-GCM-SHA384 is selected.

**Comment by chrisballinger** [comment]
Our server has no logs which makes this kind of hard to debug....

**Comment by mimi89999** [comment]
Curves are the issue here. Prosody is using secp384r1, but pubsub chatsecure is using prime256v1. That's why ECDHE isn't negotiated.

**Comment by chrisballinger** [comment]
Aha! How did you figure that out?

**Comment by mimi89999** [comment]
After @Zash said that plain RSA gets negotiated for him, I started looking why could ECDHE not get negotiated. I looked at the messaging one report and saw that different primes are used.

**Comment by chrisballinger** [comment]
Why isn't this something that is handled during cipher negotiation? Sigh..

**Comment by Zash** [comment]
It is handled, but the exact behavior has changed in OpenSSL and/or LuaSec in a way that complicates things. Before, it was treated more like how normal DH is treated; the server decides and the client just goes along with it. Now, it works more like ciphers, each party has a ordered set and something out of the intersection gets selected to be used. But due to the earlier behavior, the set of curves is just a single one so unless both ends use the exact same curve the connection either gets aborted, or it picks a different cipher suite.

**Comment by mimi89999** [comment]
@chrisballinger @Zash I resolved the issue by enabling non FS ciphersuites, but I am not happy with that...

**Comment by chrisballinger** [comment]
@mimi89999 I agree that's not ideal. I'd like to disable non-FS as well but if it breaks everyone using the wrong curve, that's a dealbreaker... :-\

**Comment by mimi89999** [comment]
>  everyone using the wrong curve

Storm about which curve is right started? 😄 

It's good to also support DH... Maybe I will install the latest version of luasec from upstream. It should support multiple curves...

**Comment by dholl** [comment]
So I added ```AES256-GCM-SHA384``` to my [ciphers list](https://check.messaging.one/result.php?id=116978#ciphers) (at the bottom...), and that fixed push notifications.

@chrisballinger - Are you the admin for pubsub.chatsecure.org?  What do you think about enabling some DHE ciphers as backups for ECDHE curve incompatibilities?  Then we could still have forward secrecy...  (Pretty please?)  :)

**Comment by dholl** [comment]
nudge nudge...  Any chance of supporting DHE ciphers on pubsub.chatsecure.org, pretty please?  :)

Feel free to put them at the bottom of your cipher list if that's your preference, but it'd be swell to provide a work around for ECDHE curve incompatibilities across the whole community of XMPP servers...

**Comment by sebastiansterk** [comment]
@chrisballinger any updates on this topic?
